x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/process/AnalyticsResultProcessor.java

```diff
@@ -107,11 +114,22 @@ public void process(AnalyticsProcess<AnalyticsResult> process) {
                 failure = "error parsing data frame analytics output: [" + e.getMessage() + "]";
             }
         } finally {
+            if (isCancelled == false) {
```

Comment:
Should we take into account `failure != null`? It seems weird to me that we are "complete" even if we failed.

Reply:
Very good point. I also took the opportunity to improve failure reporting in that class in general.